fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/provider/impl/S3Provider.java

```diff
@@ -330,23 +331,22 @@ public void close() throws Exception {
 
     @Override
     public void writeResources(String mediaType, List<ReadResultDTO> dtos) throws Exception {
-        for (ReadResultDTO dto : dtos) {
-            switch (mediaType) {
-            case FHIRMediaType.APPLICATION_PARQUET:
-                if (chunkData.getPageNum() > chunkData.getLastPageNum() || chunkData.isFinishCurrentUpload()) {
-                    pushFhirParquetToCos(dto.getResources());
-                    chunkData.setLastWritePageNum(chunkData.getPageNum());
-                }
-                break;
-            case FHIRMediaType.APPLICATION_NDJSON:
-            default:
-                if (chunkData.getPageNum() > chunkData.getLastPageNum() || chunkData.isFinishCurrentUpload()) {
-                    // TODO try PipedOutputStream -> PipedInputStream instead?  Or maybe a ByteBuffer with a flip instead?
-                    pushFhirJsonsToCos(new ByteArrayInputStream(chunkData.getBufferStream().toByteArray()), chunkData.getBufferStream().size());
-                    chunkData.setLastWritePageNum(chunkData.getPageNum());
-                }
+        switch (mediaType) {
+        case FHIRMediaType.APPLICATION_PARQUET:
+            pushFhirParquetToCos(dtos.stream()
+                    .flatMap(dto -> dto.getResources().stream())
+                    .collect(Collectors.toList()));
+            break;
+        case FHIRMediaType.APPLICATION_NDJSON:
+        default:
+            // Only if we're greater than zero, otherwise there is nothing to upload.
+            if (chunkData.getBufferStream().size() > 0) {
+                // TODO try PipedOutputStream -> PipedInputStream instead?  Or maybe a ByteBuffer with a flip instead?
+                pushFhirJsonsToCos(new ByteArrayInputStream(chunkData.getBufferStream().toByteArray()), chunkData.getBufferStream().size());
             }
+            break;
         }
+        chunkData.setLastWritePageNum(chunkData.getPageNum());
```

Comment:
this part of the change-set is more experimental in nature and should probably have been a separate PR

Reply:
ok, I backed this part out